packages/nodes-base/nodes/Discord/Discord.node.ts

```diff
@@ -23,85 +24,247 @@ export class Discord implements INodeType {
 		},
 		inputs: ['main'],
 		outputs: ['main'],
-		properties: [
+		properties:[
 			{
 				displayName: 'Webhook URL',
 				name: 'webhookUri',
 				type: 'string',
 				typeOptions: {
 					alwaysOpenEditWindow: true,
 				},
+				required: true,
 				default: '',
-				description: 'The webhook url.',
+				placeholder: 'https://discord.com/api/webhooks/ID/TOKEN',
 			},
 			{
-				displayName: 'Text',
-				name: 'text',
+				displayName: 'Content',
+				name: 'content',
 				type: 'string',
 				typeOptions: {
+					maxValue: 2000,
 					alwaysOpenEditWindow: true,
 				},
 				default: '',
-				description: 'The text to send.',
+				required: false,
+				placeholder: 'Hello World!',
+			},
+			{
+				displayName: 'Additional Fields',
+				name: 'options',
+				type: 'collection',
+				placeholder: 'Add Option',
+				default: {},
+				options: [
+					{
+						displayName: 'Username',
+						name: 'username',
+						type: 'string',
+						default: '',
+						required: false,
+						placeholder: 'User',
+					},
+					{
+						displayName: 'Components',
+						name: 'components',
+						type: 'json',
+						typeOptions: { alwaysOpenEditWindow: true, editor: 'code' },
+						default: '',
+					},
+					{
+						displayName: 'TTS',
+						name: 'tts',
+						type: 'boolean',
+						default: false,
+						required: false,
+						description: 'Should this message be sent as a Text To Speech message?',
+					},
+					{
+						displayName: 'Flags',
+						name: 'flags',
+						type: 'number',
+						default: '',
+					},
+					{
+						displayName: 'Avatar URL',
+						name: 'avatarUrl',
+						type: 'string',
+						default: '',
+						required: false,
+					},
+					{
+						displayName: 'Attachments',
+						name: 'attachments',
+						type: 'json',
+						typeOptions: { alwaysOpenEditWindow: true, editor: 'code' },
+						default: '',
+					},
+					{
+						displayName: 'Embeds',
+						name: 'embeds',
+						type: 'json',
+						typeOptions: { alwaysOpenEditWindow: true, editor: 'code' },
+						default: '',
+						required: false,
+					},
+					{
+						displayName: 'Allowed Mentions',
+						name: 'allowedMentions',
+						type: 'json',
+						typeOptions: { alwaysOpenEditWindow: true, editor: 'code' },
+						default: '',
+					},
+					{
+						displayName: 'Json Payload',
+						name: 'payloadJson',
+						type: 'json',
+						typeOptions: { alwaysOpenEditWindow: true, editor: 'code' },
+						default: '',
+					},
+				],
 			},
 		],
 	};
 
 
 
 	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const nodeInput = this.getInputData()[0].json,
+			returnData: IDataObject[] = [];
+
+		const body: DiscordWebhook = {};
+
+		const webhookUri = this.getNodeParameter('webhookUri', 0, '') as string;
```

Comment:
We should move the `webhookUri` variable as well as the `body` to inside the `for` loop since these can be expressions.